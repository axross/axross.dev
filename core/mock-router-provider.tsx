import { NextRouter } from "next/dist/next-server/lib/router/router";
import { RouterContext } from "next/dist/next-server/lib/router-context";
import * as React from "react";

export type MockRouterProviderProps = React.PropsWithChildren<NextRouter>;

/**
 *
 *
 * Do not use for production code.
 */
export function MockRouterProvider({
  pathname = "/",
  query = {},
  asPath = "/",
  push = () => Promise.resolve(false),
  replace = () => Promise.resolve(false),
  prefetch = () => Promise.resolve(undefined),
  beforePopState = () => {},
  back = () => {},
  reload = () => {},
  events: { on: eventsOn = () => {}, off: eventsOff = () => {} } = {
    on: () => {},
    off: () => {},
  } as any,
  locale = "en-US",
  locales = ["en-US", "ja-JP", "du_My"],
  defaultLocale = "en-US",
  domainLocales = [
    {
      domain: "dummy.kohei.dev",
      locales: ["du_My"],
      defaultLocale: "du_My",
    },
    {
      domain: "ja.kohei.dev",
      locales: ["ja-JP"],
      defaultLocale: "ja-JP",
    },
  ],
  isLocaleDomain = false,
  children,
}: MockRouterProviderProps) {
  const nextRouterMock = {
    pathname,
    query,
    asPath,
    isFallback: false,
    basePath: "/",
    locale,
    locales,
    defaultLocale,
    domainLocales,
    isLocaleDomain,
    push,
    replace,
    prefetch,
    beforePopState,
    back,
    reload,
    events: {
      on: eventsOn,
      off: eventsOff,
    },
  };

  return (
    <RouterContext.Provider value={nextRouterMock as any}>
      {children}
    </RouterContext.Provider>
  );
}
