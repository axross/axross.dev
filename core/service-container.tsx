import { I18nDictionaryService } from "../services/i18n-dictionary";
import { UserMonitoringService } from "../services/user-monitoring";

export interface ServiceContainer {
  i18nDictionary: I18nDictionaryService;
  userMonitoring: UserMonitoringService;
}
