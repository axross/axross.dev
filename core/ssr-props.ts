export interface CommonServerSideProps {
  locale: string;
  intlMessages: Record<string, any>;
}
