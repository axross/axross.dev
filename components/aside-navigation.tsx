import { css } from "@linaria/core";
import * as React from "react";
import { FormattedMessage } from "react-intl";
import { useRouter } from "../hooks/router";
import { LocaleSwitcher } from "./locale-switcher";
import { PostList, PostListItem } from "./post-list";
import { TableOfContents, TableOfContentsItem } from "./table-of-contents";

export interface AsideNavigationProps extends React.Attributes {
  posts: {
    slug: string;
    title: string;
  }[];
  tableOfContents: {
    id: string;
    level: number;
    text: string;
  }[];
}

export const AsideNavigation: React.VFC<AsideNavigationProps> = ({
  posts,
  tableOfContents,
  ...props
}) => {
  const { url, locale } = useRouter();

  return (
    <div {...props}>
      <div
        className={css`
          padding-inline-start: var(--space-md);
          padding-inline-end: var(--space-md);
        `}
      >
        <LocaleSwitcher
          className={css`
            justify-content: flex-end;
          `}
        />
      </div>

      <PostList
        className={css`
          margin-block-start: var(--space-xl);
        `}
      >
        <PostListItem
          href="/"
          as="/"
          tableOfContents={
            url.pathname === `/${locale}` ? (
              <TableOfContents data-testid="table-of-contents">
                {tableOfContents.map(({ id, level, text }) => (
                  <TableOfContentsItem
                    targetId={id}
                    level={level}
                    data-testid="item"
                    key={id}
                  >
                    {text}
                  </TableOfContentsItem>
                ))}
              </TableOfContents>
            ) : undefined
          }
        >
          <FormattedMessage defaultMessage="About" />
        </PostListItem>

        {posts.map((p) => (
          <PostListItem
            href="/posts/[slug]"
            as={`/posts/${p.slug}`}
            tableOfContents={
              url.pathname === `/${locale}/posts/${p.slug}` ? (
                <TableOfContents data-testid="table-of-contents">
                  {tableOfContents.map(({ id, level, text }) => (
                    <TableOfContentsItem
                      targetId={id}
                      level={level}
                      data-testid="item"
                      key={id}
                    >
                      {text}
                    </TableOfContentsItem>
                  ))}
                </TableOfContents>
              ) : undefined
            }
            key={p.slug}
          >
            {p.title}
          </PostListItem>
        ))}
      </PostList>
    </div>
  );
};
