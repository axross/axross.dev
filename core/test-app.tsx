import { NextRouter } from "next/dist/next-server/lib/router/router";
import mitt from "next/dist/next-server/lib/mitt";
import { RouterContext } from "next/dist/next-server/lib/router-context";
import * as React from "react";
import { IntlProvider } from "react-intl";
import { ServiceProvider } from "../components/service";
import { OriginProvider } from "../global-hooks/url";
import { EmptyI18nDictionaryService } from "../services/i18n-dictionary";
import { EmptyUserMonitoringService } from "../services/user-monitoring";
import { ServiceContainer } from "./service-container";

interface TestAppProps extends React.Attributes {
  origin?: string;
  router?: {
    pathname?: string;
    query?: Record<string, string | string[]>;
    asPath?: string;
    push?: (...params: Parameters<NextRouter["push"]>) => void;
    replace?: (...params: Parameters<NextRouter["replace"]>) => void;
    prefetch?: (...params: Parameters<NextRouter["prefetch"]>) => void;
    beforePopState?: (
      ...params: Parameters<NextRouter["beforePopState"]>
    ) => void;
    back?: (...params: Parameters<NextRouter["back"]>) => void;
    reload?: (...params: Parameters<NextRouter["reload"]>) => void;
  };
  intl?: {
    messages?: Record<string, any>;
    locale?: string;
    defaultLocale?: string;
  };
  serviceContainer?: Partial<ServiceContainer>;
}

export interface TestAppImperativeHandle {
  emulateRouteChangeStart: (...args: any[]) => void;
  emulateRouteChangeComplete: (...args: any[]) => void;
  emulateRouteChangeError: (...args: any[]) => void;
  emulateBeforeHistoryChange: (...args: any[]) => void;
  emulateHashChangeStart: (...args: any[]) => void;
  emulateHashChangeComplete: (...args: any[]) => void;
}

export type TestAppElement = TestAppImperativeHandle;

export const TestApp = React.forwardRef<
  TestAppElement,
  React.PropsWithChildren<TestAppProps>
>(
  (
    {
      origin = "https://dummy.kohei.dev",
      router: {
        pathname = "/",
        query = {},
        asPath = "/",
        push = () => {},
        replace = () => {},
        prefetch = () => {},
        beforePopState = () => {},
        back = () => {},
        reload = () => {},
      } = {},
      intl: { messages = {}, locale = "en-US", defaultLocale = "en-US" } = {},
      serviceContainer: {
        i18nDictionary = new EmptyI18nDictionaryService(),
        userMonitoring = new EmptyUserMonitoringService(),
      } = {},
      children,
    },
    ref
  ) => {
    const nextRouterMockMutRef = React.useRef<NextRouter>();
    const nextRouterMock = React.useMemo(() => {
      if (!nextRouterMockMutRef.current) {
        const events = mitt();
        nextRouterMockMutRef.current = {
          pathname,
          query,
          asPath,
          push: (...args) => {
            push(...args);

            return Promise.resolve(false);
          },
          replace: (...args) => {
            replace(...args);

            return Promise.resolve(false);
          },
          prefetch: (...args) => {
            prefetch(...args);

            return Promise.resolve(undefined);
          },
          beforePopState,
          back,
          reload,
          route: "",
          basePath: "",
          events,
          isLocaleDomain: false,
          isFallback: false,
          isReady: true,
          isPreview: false,
        };
      }

      return nextRouterMockMutRef.current;
    }, [
      pathname,
      query,
      asPath,
      push,
      replace,
      prefetch,
      beforePopState,
      back,
      reload,
    ]);

    React.useImperativeHandle<TestAppImperativeHandle, TestAppImperativeHandle>(
      ref,
      () => ({
        emulateRouteChangeStart: (...args: any[]) =>
          nextRouterMockMutRef?.current?.events.emit(
            "routeChangeStart",
            ...args
          ),
        emulateRouteChangeComplete: (...args: any[]) =>
          nextRouterMockMutRef?.current?.events.emit(
            "routeChangeComplete",
            ...args
          ),
        emulateRouteChangeError: (...args: any[]) =>
          nextRouterMockMutRef?.current?.events.emit(
            "routeChangeError",
            ...args
          ),
        emulateBeforeHistoryChange: (...args: any[]) =>
          nextRouterMockMutRef?.current?.events.emit(
            "beforeHistoryChange",
            ...args
          ),
        emulateHashChangeStart: (...args: any[]) =>
          nextRouterMockMutRef?.current?.events.emit(
            "hashChangeStart",
            ...args
          ),
        emulateHashChangeComplete: (...args: any[]) =>
          nextRouterMockMutRef?.current?.events.emit(
            "hashChangeComplete",
            ...args
          ),
      }),
      []
    );

    return (
      <RouterContext.Provider value={nextRouterMock}>
        <ServiceProvider serviceContainer={{ i18nDictionary, userMonitoring }}>
          <OriginProvider origin={origin}>
            <IntlProvider
              messages={messages}
              locale={locale}
              defaultLocale={defaultLocale}
            >
              {children}
            </IntlProvider>
          </OriginProvider>
        </ServiceProvider>
      </RouterContext.Provider>
    );
  }
);
