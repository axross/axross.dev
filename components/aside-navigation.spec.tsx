import { render } from "@testing-library/react";
import * as React from "react";
import { TestApp } from "../core/test-app";
import { AsideNavigation } from "./aside-navigation";

const posts = [
  { slug: "47qwqp", title: "aliquid autem eius et non iure vitae" },
  { slug: "lccd43", title: "officiis alias est amet illo saepe omnis" },
  { slug: "mrrw0e", title: "fugiat delectus aut voluptatem beatae ab" },
  { slug: "csvh21", title: "fugit recusandae aut veniam eos iure ut quia" },
  { slug: "pn04ii", title: "dolores laudantium est voluptates ea odio labore" },
  { slug: "6qem31", title: "doloribus" },
  { slug: "r5ylk0", title: "vel maiores id sapiente voluptatem" },
  { slug: "znq279", title: "libero" },
  { slug: "sg4g6o", title: "porro at exercitationem officiis illum molestias" },
  { slug: "sz3nty", title: "non perspiciatis est magnam" },
];

const tableOfContents = [
  { id: "zg2r6g", level: 2, text: "in aut natus eius nesciunt" },
  { id: "2sfiwl", level: 3, text: "et quidem vel labore aut vero eaque" },
  { id: "blmqmw", level: 3, text: "aut omnis saepe totam aut quasi magni" },
  {
    id: "gg51id",
    level: 4,
    text:
      "consequatur reprehenderit aut qui voluptatum tempora qui fugiat accusamus",
  },
  { id: "jzmtod", level: 3, text: "qui quia quo sunt" },
  { id: "1lpeq1", level: 4, text: "est in inventore rerum" },
  { id: "bf4cdf", level: 2, text: "voluptatem labore" },
  { id: "gu0a2v", level: 2, text: "odit voluptatum optio possimus alias aut" },
  { id: "lcm25k", level: 3, text: "quia facilis tenetur velit aliquam atque" },
  { id: "xw1lzt", level: 4, text: "rerum molestias hic harum at placeat" },
  { id: "lwzd8q", level: 5, text: "accusamus eius aspernatur labore nulla" },
];

describe("<AsideNavigation>", () => {
  it("renders the list of all posts and the table of contents of the article currently shown (when the user is at index page)", () => {
    const { getByTestId } = render(
      <TestApp
        router={{
          pathname: "/",
          asPath: "/",
        }}
      >
        <AsideNavigation
          posts={posts}
          tableOfContents={tableOfContents}
          data-testid="aside-navigation"
        />
      </TestApp>
    );

    expect(getByTestId("aside-navigation")).toMatchSnapshot();
  });

  it("renders the list of all posts and the table of contents of the article currently shown (when the user is at post page)", () => {
    const { getByTestId } = render(
      <TestApp
        router={{
          pathname: "/[locale]/posts/[slug]",
          asPath: "/en-us/posts/b4a2a0c1-41d9-4e03-aec1-20b57358617a",
        }}
      >
        <AsideNavigation
          posts={posts}
          tableOfContents={tableOfContents}
          data-testid="aside-navigation"
        />
      </TestApp>
    );

    expect(getByTestId("aside-navigation")).toMatchSnapshot();
  });
});
