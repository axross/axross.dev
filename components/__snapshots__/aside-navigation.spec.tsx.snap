// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AsideNavigation> renders the list of all posts and the table of contents of the article currently shown (when the user is at index page) 1`] = `
<div
  data-testid="aside-navigation"
>
  <div
    class="dinqm8k"
  >
    <ul
      class="u16duzwh lucqacz"
    >
      <li>
        <a
          class="a1v4h6uj"
          data-testid="link"
          href="/en-us"
          title="en-us"
        >
          <svg
            class="f1mxy9mn"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="256"
              cy="256"
              fill="#f0f0f0"
              r="256"
            />
            <g
              fill="#d80027"
            >
              <path
                d="M244.87 256H512c0-23.106-3.08-45.49-8.819-66.783H244.87V256zM244.87 122.435h229.556a257.35 257.35 0 00-59.07-66.783H244.87v66.783zM256 512c60.249 0 115.626-20.824 159.356-55.652H96.644C140.374 491.176 195.751 512 256 512zM37.574 389.565h436.852a254.474 254.474 0 0028.755-66.783H8.819a254.474 254.474 0 0028.755 66.783z"
              />
            </g>
            <path
              d="M118.584 39.978h23.329l-21.7 15.765 8.289 25.509-21.699-15.765-21.699 15.765 7.16-22.037a257.407 257.407 0 00-49.652 55.337h7.475l-13.813 10.035a255.58 255.58 0 00-6.194 10.938l6.596 20.301-12.306-8.941a253.567 253.567 0 00-8.372 19.873l7.267 22.368h26.822l-21.7 15.765 8.289 25.509-21.699-15.765-12.998 9.444A258.468 258.468 0 000 256h256V0c-50.572 0-97.715 14.67-137.416 39.978zm9.918 190.422l-21.699-15.765L85.104 230.4l8.289-25.509-21.7-15.765h26.822l8.288-25.509 8.288 25.509h26.822l-21.7 15.765 8.289 25.509zm-8.289-100.083l8.289 25.509-21.699-15.765-21.699 15.765 8.289-25.509-21.7-15.765h26.822l8.288-25.509 8.288 25.509h26.822l-21.7 15.765zM220.328 230.4l-21.699-15.765L176.93 230.4l8.289-25.509-21.7-15.765h26.822l8.288-25.509 8.288 25.509h26.822l-21.7 15.765 8.289 25.509zm-8.289-100.083l8.289 25.509-21.699-15.765-21.699 15.765 8.289-25.509-21.7-15.765h26.822l8.288-25.509 8.288 25.509h26.822l-21.7 15.765zm0-74.574l8.289 25.509-21.699-15.765-21.699 15.765 8.289-25.509-21.7-15.765h26.822l8.288-25.509 8.288 25.509h26.822l-21.7 15.765z"
              fill="#0052b4"
            />
          </svg>
        </a>
      </li>
      <li>
        <a
          class="a1v4h6uj"
          data-testid="link"
          href="/ja-jp"
          title="ja-jp"
        >
          <svg
            class="f1mxy9mn"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="256"
              cy="256"
              fill="#f0f0f0"
              r="256"
            />
            <circle
              cx="256"
              cy="256"
              fill="#d80027"
              r="111.304"
            />
          </svg>
        </a>
      </li>
    </ul>
  </div>
  <nav
    class="n1cyut43 p1sv5lrk"
  >
    <ul
      class="ugmswd7"
    >
      <li
        class="lc47www"
      >
        <a
          class="ahqgba"
          href="/en-us"
        >
          <h2
            class="hah7idx hk3fxxj"
          >
            About
          </h2>
        </a>
      </li>
      <li
        class="lc47www"
      >
        <a
          class="ahqgba"
          href="/en-us/posts/47qwqp"
        >
          <h2
            class="hah7idx hk3fxxj"
          >
            aliquid autem eius et non iure vitae
          </h2>
        </a>
      </li>
      <li
        class="lc47www"
      >
        <a
          class="ahqgba"
          href="/en-us/posts/lccd43"
        >
          <h2
            class="hah7idx hk3fxxj"
          >
            officiis alias est amet illo saepe omnis
          </h2>
        </a>
      </li>
      <li
        class="lc47www"
      >
        <a
          class="ahqgba"
          href="/en-us/posts/mrrw0e"
        >
          <h2
            class="hah7idx hk3fxxj"
          >
            fugiat delectus aut voluptatem beatae ab
          </h2>
        </a>
      </li>
      <li
        class="lc47www"
      >
        <a
          class="ahqgba"
          href="/en-us/posts/csvh21"
        >
          <h2
            class="hah7idx hk3fxxj"
          >
            fugit recusandae aut veniam eos iure ut quia
          </h2>
        </a>
      </li>
      <li
        class="lc47www"
      >
        <a
          class="ahqgba"
          href="/en-us/posts/pn04ii"
        >
          <h2
            class="hah7idx hk3fxxj"
          >
            dolores laudantium est voluptates ea odio labore
          </h2>
        </a>
      </li>
      <li
        class="lc47www"
      >
        <a
          class="ahqgba"
          href="/en-us/posts/6qem31"
        >
          <h2
            class="hah7idx hk3fxxj"
          >
            doloribus
          </h2>
        </a>
      </li>
      <li
        class="lc47www"
      >
        <a
          class="ahqgba"
          href="/en-us/posts/r5ylk0"
        >
          <h2
            class="hah7idx hk3fxxj"
          >
            vel maiores id sapiente voluptatem
          </h2>
        </a>
      </li>
      <li
        class="lc47www"
      >
        <a
          class="ahqgba"
          href="/en-us/posts/znq279"
        >
          <h2
            class="hah7idx hk3fxxj"
          >
            libero
          </h2>
        </a>
      </li>
      <li
        class="lc47www"
      >
        <a
          class="ahqgba"
          href="/en-us/posts/sg4g6o"
        >
          <h2
            class="hah7idx hk3fxxj"
          >
            porro at exercitationem officiis illum molestias
          </h2>
        </a>
      </li>
      <li
        class="lc47www"
      >
        <a
          class="ahqgba"
          href="/en-us/posts/sz3nty"
        >
          <h2
            class="hah7idx hk3fxxj"
          >
            non perspiciatis est magnam
          </h2>
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`<AsideNavigation> renders the list of all posts and the table of contents of the article currently shown (when the user is at post page) 1`] = `
<div
  data-testid="aside-navigation"
>
  <div
    class="dinqm8k"
  >
    <ul
      class="u16duzwh lucqacz"
    >
      <li>
        <a
          class="a1v4h6uj"
          data-testid="link"
          href="/en-us/posts/b4a2a0c1-41d9-4e03-aec1-20b57358617a"
          title="en-us"
        >
          <svg
            class="f1mxy9mn"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="256"
              cy="256"
              fill="#f0f0f0"
              r="256"
            />
            <g
              fill="#d80027"
            >
              <path
                d="M244.87 256H512c0-23.106-3.08-45.49-8.819-66.783H244.87V256zM244.87 122.435h229.556a257.35 257.35 0 00-59.07-66.783H244.87v66.783zM256 512c60.249 0 115.626-20.824 159.356-55.652H96.644C140.374 491.176 195.751 512 256 512zM37.574 389.565h436.852a254.474 254.474 0 0028.755-66.783H8.819a254.474 254.474 0 0028.755 66.783z"
              />
            </g>
            <path
              d="M118.584 39.978h23.329l-21.7 15.765 8.289 25.509-21.699-15.765-21.699 15.765 7.16-22.037a257.407 257.407 0 00-49.652 55.337h7.475l-13.813 10.035a255.58 255.58 0 00-6.194 10.938l6.596 20.301-12.306-8.941a253.567 253.567 0 00-8.372 19.873l7.267 22.368h26.822l-21.7 15.765 8.289 25.509-21.699-15.765-12.998 9.444A258.468 258.468 0 000 256h256V0c-50.572 0-97.715 14.67-137.416 39.978zm9.918 190.422l-21.699-15.765L85.104 230.4l8.289-25.509-21.7-15.765h26.822l8.288-25.509 8.288 25.509h26.822l-21.7 15.765 8.289 25.509zm-8.289-100.083l8.289 25.509-21.699-15.765-21.699 15.765 8.289-25.509-21.7-15.765h26.822l8.288-25.509 8.288 25.509h26.822l-21.7 15.765zM220.328 230.4l-21.699-15.765L176.93 230.4l8.289-25.509-21.7-15.765h26.822l8.288-25.509 8.288 25.509h26.822l-21.7 15.765 8.289 25.509zm-8.289-100.083l8.289 25.509-21.699-15.765-21.699 15.765 8.289-25.509-21.7-15.765h26.822l8.288-25.509 8.288 25.509h26.822l-21.7 15.765zm0-74.574l8.289 25.509-21.699-15.765-21.699 15.765 8.289-25.509-21.7-15.765h26.822l8.288-25.509 8.288 25.509h26.822l-21.7 15.765z"
              fill="#0052b4"
            />
          </svg>
        </a>
      </li>
      <li>
        <a
          class="a1v4h6uj"
          data-testid="link"
          href="/ja-jp/posts/b4a2a0c1-41d9-4e03-aec1-20b57358617a"
          title="ja-jp"
        >
          <svg
            class="f1mxy9mn"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="256"
              cy="256"
              fill="#f0f0f0"
              r="256"
            />
            <circle
              cx="256"
              cy="256"
              fill="#d80027"
              r="111.304"
            />
          </svg>
        </a>
      </li>
    </ul>
  </div>
  <nav
    class="n1cyut43 p1sv5lrk"
  >
    <ul
      class="ugmswd7"
    >
      <li
        class="lc47www"
      >
        <a
          class="ahqgba"
          href="/en-us"
        >
          <h2
            class="hah7idx hk3fxxj"
          >
            About
          </h2>
        </a>
      </li>
      <li
        class="lc47www"
      >
        <a
          class="ahqgba"
          href="/en-us/posts/47qwqp"
        >
          <h2
            class="hah7idx hk3fxxj"
          >
            aliquid autem eius et non iure vitae
          </h2>
        </a>
      </li>
      <li
        class="lc47www"
      >
        <a
          class="ahqgba"
          href="/en-us/posts/lccd43"
        >
          <h2
            class="hah7idx hk3fxxj"
          >
            officiis alias est amet illo saepe omnis
          </h2>
        </a>
      </li>
      <li
        class="lc47www"
      >
        <a
          class="ahqgba"
          href="/en-us/posts/mrrw0e"
        >
          <h2
            class="hah7idx hk3fxxj"
          >
            fugiat delectus aut voluptatem beatae ab
          </h2>
        </a>
      </li>
      <li
        class="lc47www"
      >
        <a
          class="ahqgba"
          href="/en-us/posts/csvh21"
        >
          <h2
            class="hah7idx hk3fxxj"
          >
            fugit recusandae aut veniam eos iure ut quia
          </h2>
        </a>
      </li>
      <li
        class="lc47www"
      >
        <a
          class="ahqgba"
          href="/en-us/posts/pn04ii"
        >
          <h2
            class="hah7idx hk3fxxj"
          >
            dolores laudantium est voluptates ea odio labore
          </h2>
        </a>
      </li>
      <li
        class="lc47www"
      >
        <a
          class="ahqgba"
          href="/en-us/posts/6qem31"
        >
          <h2
            class="hah7idx hk3fxxj"
          >
            doloribus
          </h2>
        </a>
      </li>
      <li
        class="lc47www"
      >
        <a
          class="ahqgba"
          href="/en-us/posts/r5ylk0"
        >
          <h2
            class="hah7idx hk3fxxj"
          >
            vel maiores id sapiente voluptatem
          </h2>
        </a>
      </li>
      <li
        class="lc47www"
      >
        <a
          class="ahqgba"
          href="/en-us/posts/znq279"
        >
          <h2
            class="hah7idx hk3fxxj"
          >
            libero
          </h2>
        </a>
      </li>
      <li
        class="lc47www"
      >
        <a
          class="ahqgba"
          href="/en-us/posts/sg4g6o"
        >
          <h2
            class="hah7idx hk3fxxj"
          >
            porro at exercitationem officiis illum molestias
          </h2>
        </a>
      </li>
      <li
        class="lc47www"
      >
        <a
          class="ahqgba"
          href="/en-us/posts/sz3nty"
        >
          <h2
            class="hah7idx hk3fxxj"
          >
            non perspiciatis est magnam
          </h2>
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
